### PR TITLE
[Xcodeproj] More functional tests

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
* Adds the C and Swift mixed package fixture to Xcodeproj functional tests.

* SR-1444:
for a target A-B swift build creates module name is AB where as Xcode as A_B. I am not sure which one should be correct but using c99name for target names solves the problem in Xcode.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Dependant on #335

**Comment by aciidb0mb3r** [comment]
@ddunbar Added a pkgconfig regression test

**Comment by ddunbar** [comment]
Can you rebase post merge? I unintentionally did a squash merge.

**Comment by aciidb0mb3r** [comment]
@ddunbar done

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

File: Sources/Xcodeproj/pbxproj().swift [hunk 1]
```diff
@@ -94,7 +94,7 @@ public func pbxproj(srcroot: String, projectRoot: String, xcodeprojPath: String,
         print("            buildPhases = (\(module.compilePhaseReference), \(module.linkPhaseReference));")
         print("            buildRules = ();")
         print("            dependencies = (\(module.nativeTargetDependencies));")
-        print("            name = \(module.name);")
+        print("            name = \(module.c99name);")
```
Comment (ddunbar) [inline Sources/Xcodeproj/pbxproj().swift]: I don't think this is what we want, did this fix a problem you encountered on one of the tests?

Comment (aciidb0mb3r) [inline Sources/Xcodeproj/pbxproj().swift]: yeah, see the testcase in this commit and SR-1444 also: 
for a target A-B swift build creates module name is AB where as Xcode as A_B. I am not sure which one should be correct but using c99name for target names solves the problem in Xcode.

Comment (ddunbar) [inline Sources/Xcodeproj/pbxproj().swift]: SwiftPM is getting it wrong, then. We should match Xcode in our computation of module names... Xcode substitutes invalid characters with `_`.

Comment (aciidb0mb3r) [inline Sources/Xcodeproj/pbxproj().swift]: cool

**Pull request merged**